### PR TITLE
Adds SparkFun Pro Micro C3 Variant File

### PR DESCRIPTION
**Pull request by edspark**
## Description of Change
Adds the SparkFun Pro Micro C3 Variant File and adds the board to boards.txt.

## Tests scenarios
Tested with v2.0.11 of the Espressif ESP32 Arduino Core.

## Related links
This is the related [product](https://www.sparkfun.com/products/23484)

### CODE REVIEW

File: variants/sparkFun_pro_micro_esp32c3/pins_arduino.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#define USB_SERIAL         "" // Empty string for MAC adddress
+
+
+#define EXTERNAL_NUM_INTERRUPTS 22
+#define NUM_DIGITAL_PINS        22
+#define NUM_ANALOG_INPUTS       6
+
+#define analogInputToDigitalPin(p)  (((p)<NUM_ANALOG_INPUTS)?(analogChannelToDigitalPin(p)):-1)
+#define digitalPinToInterrupt(p)    (((p)<NUM_DIGITAL_PINS)?(p):-1)
+#define digitalPinHasPWM(p)         (p < EXTERNAL_NUM_INTERRUPTS)
```
Comment (P-R-O-C-H-Y) [inline variants/sparkFun_pro_micro_esp32c3/pins_arduino.h:19]: This part was moved to `Arduino.h` file, as it common for each SoC. Please remove it.

File: variants/sparkFun_pro_micro_esp32c3/pins_arduino.h [hunk 2]
```diff
@@ -0,0 +1,58 @@
+#ifndef Pins_Arduino_h
```
Comment (P-R-O-C-H-Y) [inline variants/sparkFun_pro_micro_esp32c3/pins_arduino.h:1]: Fix variant folder name to lowercase.

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@edspark PTAL on my comments.

File: variants/sparkFun_pro_micro_esp32c3/pins_arduino.h (hunk 1 above)
Comment (edspark) [reply variants/sparkFun_pro_micro_esp32c3/pins_arduino.h]: Fixed!

File: variants/sparkFun_pro_micro_esp32c3/pins_arduino.h (hunk 2 above)
Comment (edspark) [reply variants/sparkFun_pro_micro_esp32c3/pins_arduino.h:1]: Fixed!

Comment (P-R-O-C-H-Y) [reply variants/sparkFun_pro_micro_esp32c3/pins_arduino.h:1]: Compilation still fails. The folder name was not changed in the last commit. PTAL @edspark

Comment (edspark) [reply variants/sparkFun_pro_micro_esp32c3/pins_arduino.h]: Actually fixed this time, I'm not sure why it didn't take the first time.

**Review by P-R-O-C-H-Y — APPROVED** [review]

**Pull request merged**